src/routes/safe/components/Balances/SendModal/screens/ContractInteraction/ContractABI/index.tsx

```diff
@@ -19,12 +23,40 @@ const mustBeValidABI = (abi: string): undefined | string => {
   }
 }
 
-const ContractABI = () => (
-  <Row margin="sm">
-    <Col>
-      <TextareaField name="abi" placeholder="ABI*" text="ABI*" type="text" validate={mustBeValidABI} />
-    </Col>
-  </Row>
-)
+const ContractABI = (): React.ReactElement => {
+  const {
+    input: { value: contractAddress },
+  } = useField('contractAddress', { subscription: { value: true } })
+  const { mutators } = useForm()
+  const setAbiValue = React.useRef(mutators.setAbiValue)
+
+  React.useEffect(() => {
+    const validateAndSetAbi = async () => {
+      if (!mustBeEthereumAddress(contractAddress) && !(await mustBeEthereumContractAddress(contractAddress))) {
+        const network = getNetwork()
+        const source = getConfiguredSource()
+        const abi = await source.getContractABI(contractAddress, network)
+
+        // this check may help in scenarios where the user first pastes the ABI,
```

Comment:
should we notify the users of this functionality in the UI? Perhaps if we detect that the ABI it's a proxy, we can show a label saying that he can replace the address and the ABI won't be changed.

Reply:
not sure about this, I have no certainty that all the proxy contracts won't have a callable method